pyanaconda/modules/network/network.py

```diff
@@ -35,6 +39,10 @@ def __init__(self):
         super().__init__()
         self.hostname_changed = Signal()
         self._hostname = "localhost.localdomain"
+        self.current_hostname_changed = Signal()
+        # TODO fallback solution (no hostnamed) ?
```

Comment:
I imagine setting Environment property (eg Anaconda, InitialSetup, LiveCD, System, Mock) of the module and make the module behave depending on it. For example either use hostnamed or use socket module or be noop. So do not try to connect to hostnamed service unconditionally like we do here.